src/AutoRest.CSharp/Common/Input/Utf8JsonReaderExtensions.cs

```diff
@@ -3,14 +3,31 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 
 namespace AutoRest.CSharp.Common.Input
 {
     internal static class Utf8JsonReaderExtensions
     {
+        public static bool TryReadReferenceId(this ref Utf8JsonReader reader, ref string? id)
```

Comment:
why is this needed? can it be replaced with the below method of `isFirstProperty = true`?

Reply:
this is only a result of copying code from MTG.
I think it is the other version of this method that we do not need.
In our current usage, we did not really try to verify if `$id` is the first property we met, we just assume this way.

Reply:
ok, then the duplication is in MTG in the first place. We can clean it up later there instead.

Reply:
yeah I am planning to remove the one with `isFirstProperty` in a near future